plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.plugin
+
+import com.ritense.plugin.annotation.Plugin
+import com.ritense.plugin.domain.PluginDefinition
+import com.ritense.plugin.repository.PluginDefinitionRepository
+import org.springframework.boot.context.event.ApplicationStartedEvent
+import org.springframework.context.event.EventListener
+
+class PluginDeploymentListener(
+    private val pluginDefinitionResolver: PluginDefinitionResolver,
+    private val pluginDefinitionRepository: PluginDefinitionRepository
+) {
+
+    @EventListener(ApplicationStartedEvent::class)
+    fun deployPluginDefinitions() {
+        val classes = findPluginClasses()
+        val action: (Map.Entry<Class<*>, Plugin>) -> Unit = { (clazz, pluginAnnotation) ->
```

Comment:
I feel like "action" is a bit of a loaded term. maybe rename this variable (or just omit naming this variable altogether, I don't think this matters much for readability).

Reply:
Agreed, action is more a term used for the PluginAction (methods)

Reply:
fixed